environments/templates/next-visit-fan-out-application.yaml

```diff
@@ -0,0 +1,28 @@
+{{- if (index .Values "applications"  "prompt-proto-service-hsc") -}}
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: next-visit-fan-out
+  namespace: argocd
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io
+spec:
+  syncPolicy:
+    managedNamespaceMetadata:
+      labels:
+        argocd.argoproj.io/managed-by: argocd
+    syncOptions:
+    - CreateNamespace=true
```

Comment:
This is interesting and we probably should think about switching over to it, but to keep things consistent for now, could you explicitly manage the namespace as a resource the way that the other applications do?